### ISSUE TITLE
PySINDy simulation issues

### ISSUE THREAD

**Issue by joardima** [dynamicslab/pysindy#275]
Dear Sirs.

I'm trying to use SINDy method, through PySINDy package, to obtaining equations of a water distribution network for prediction of pressure (P), demand (D), and flow (Q) variables.

Some information of the input data used is presented next:

- The training data consists on measurements of P, D, and Q during 241 timesteps (24 hours @ 6 minutes or 360 seconds; defined in the variable **_DT_**)
- The 29 features consists of the following elements, defined in the variable **_FEATURE_NAMES_**:
PJ_1 to PJ_6 and PR_7
D_1 to D_6
Q_1 to Q_8 
- A total of 241 timesteps (24 hours @ 6 minutes or 360 seconds) are defined in the array **_TS_**
- The measured data is defined in the variable _**X_TRAIN**_ of size 241 x 29 (number of timesteps x number of variables/features)
- Lambda value equals to 0.01
- Alpha value equals to 0.05
- Polynomial Library degree equals 1
- Polynomial Library include interaction equals False
- Maximum number of iterations equals to 20
- Optimizer equals to STLSQ

**APPROACH CASES**

**Case 1: x equals to X_TRAIN**
- PySINDy object is created using _x equals to X_TRAIN_ and stored in the variable PySINDy_OBJ
- PySINDy equations are obtained using _PySINDy_OBJ.equations_ , stored in the variable PySINDy_EQU, included in the below picture ,and attached in the file '_[Case_01_Equations.xlsx](https://github.com/dynamicslab/pysindy/files/10342459/Case_01_Equations.xlsx)_'.
 
![Case_01_Equations](https://user-images.githubusercontent.com/97732873/210512446-6f60fefb-0248-4598-aa7f-b7e69da51fc3.png)

- PySINDy simulation using _PySINDy_SIM = PySINDy_OBJ.simulate(x0 = X_TRAIN[0], t = TS)_ gives no results, instead the following message is presented '_ValueError: Input contains infinity or a value too large for dtype('float64'). capi_return is NULL Call-back cb_f_in_lsoda__user__routines failed._'
- Code for Case 1 file: '_Case_01_code.py_' within the file '_[Codes.zip](https://github.com/dynamicslab/pysindy/files/10343171/Codes.zip)_'

**Case 2: defining X and X_DOT as subsets of X_TRAIN**
- PySINDy object is created using _X       = X_TRAIN[:-1,:] and X_DOT   = X_TRAIN[1:,:]_ and stored in the variable PySINDy_OBJ
- PySINDy equations are obtained using _PySINDy_OBJ.equations_ , stored in the variable PySINDy_EQU, included in the below picture ,and attached in the file '_[Case_02_Equations.xlsx](https://github.com/dynamicslab/pysindy/files/10342606/Case_02_Equations.xlsx)_'.

![Case_02_Equations](https://user-images.githubusercontent.com/97732873/210516481-e02a1f3a-7b42-4ac3-9f6c-3c2fbfe05802.png)

- PySINDy simulation using _PySINDy_SIM = PySINDy_OBJ.simulate(x0 = X_TRAIN[0], t = TS)_ gives no results, instead the following message is presented '_ValueError: Input contains infinity or a value too large for dtype('float64'). capi_return is NULL Call-back cb_f_in_lsoda__user__routines failed._'
- Code for Case 2 file: '_Case_02_code.py_' within the file '_[Codes.zip](https://github.com/dynamicslab/pysindy/files/10343171/Codes.zip)_'

**QUESTIONS**

1. How PySINDy predict values when the resulting PySINDy objects have zero coefficients for the features in the resulting equations?
2. What could I change in order to obtain results when trying to simulate the resulting SINDy models using _PySINDy_OBJ.simulate_ method?

**N.B.:** I attached the codes as a .zip file and not in the body enclose in triple backticks (```) because the maximum length allowed.

**Comment by Jacob-Stevens-Haas** [comment]
Hey, this is a really cool use case - in particular, I love the idea of using pysindy for industrial systems.

It would help if you wrote this as a minimal working example of code I could copy and paste.  In particular, I'm not sure where the $x^{1.825}$ terms are coming from - they shouldn't be in a `PolynomialLibrary` object.  Secondly, it looks like you're defining `x_dot` as the "next value" of `x_train`.  In that case, you need to set `discrete=True` to tell pysindy that x_dot are not derivatives themselves.

**Comment by Jacob-Stevens-Haas** [comment]
I made a few changes and this worked well:
* Remove the Q^1.852 variables.  It looks like you're trying to use a custom library by pre-calculating features rather than setting up a library to handle this.  That can work, but unless there's something about the 1.852 that you know *a priori*, just delete these variables.
* Sparse Optimizer: I switched back to the default values for `threshold` and `alpha`.  I also set `normalize_columns` to `True`, since your data is of very different scales
* I also manually rescaled all the data to the same magnitude.  The `PJ` quantities were order 10^4 larger than the `D` and `Q` quantities, so I multiplied the latter by 10^4.  This will affect the RMSE values for those quantities by including that 10^4 conversion factor.
